libbeat/cfgfile/cfgfile.go

```diff
@@ -133,30 +134,36 @@ func Read(out interface{}, path string) error {
 // Load reads the configuration from a YAML file structure. If path is empty
 // this method reads from the configuration file specified by the '-c' command
 // line flag.
+// This function cares about the underlying fleet setting, and if beats is running with
+// the management.enabled flag, Load() will bypass reading a config file, and merely merge any overrides.
 func Load(path string, beatOverrides []ConditionalOverride) (*config.C, error) {
 	var c *config.C
 	var err error
 
 	cfgpath := GetPathConfig()
 
-	if path == "" {
-		list := []string{}
-		for _, cfg := range configfiles.List() {
-			if !filepath.IsAbs(cfg) {
-				list = append(list, filepath.Join(cfgpath, cfg))
-			} else {
-				list = append(list, cfg)
+	if !fleetmode.Enabled() {
+		if path == "" {
+			list := []string{}
+			for _, cfg := range configfiles.List() {
+				if !filepath.IsAbs(cfg) {
+					list = append(list, filepath.Join(cfgpath, cfg))
+				} else {
+					list = append(list, cfg)
+				}
+			}
+			c, err = common.LoadFiles(list...)
+		} else {
+			if !filepath.IsAbs(path) {
+				path = filepath.Join(cfgpath, path)
 			}
+			c, err = common.LoadFile(path)
 		}
-		c, err = common.LoadFiles(list...)
-	} else {
-		if !filepath.IsAbs(path) {
-			path = filepath.Join(cfgpath, path)
+		if err != nil {
+			return nil, err
 		}
-		c, err = common.LoadFile(path)
-	}
-	if err != nil {
-		return nil, err
+	} else {
+		c = config.NewConfig()
```

Comment:
APM server is relying on configuration overrides: https://github.com/elastic/beats/issues/31901#issuecomment-1225755605

Can you confirm that overrides like https://github.com/elastic/apm-server/blob/main/internal/beatcmd/root.go#L42-L68 will still work without changes to the code that depends on them?

Reply:
Yup, I wrote this such that we wouldn't bypass any overrides set by the beat.